src_py/sprite.py

```diff
@@ -1040,131 +1066,137 @@ def draw(self, surface, bgd=None):
-
-                        _spr_rect_clip = _spr_rect.clip
-
-                        for idx in _spr_rect.collidelistall(_update):
-                            # clip
-                            clip = _spr_rect_clip(_update[idx])
-                            _surf_blit(spr.image,
-                                       clip,
-                                       (clip[0] + rect_offset_x,
-                                        clip[1] + rect_offset_y,
-                                        clip[2],
-                                        clip[3]),
-                                       spr.blendmode)
-                else: # dirty sprite
-                    if spr._visible:
-                        _old_rect[spr] = _surf_blit(spr.image,
-                                                    spr.rect,
-                                                    spr.source_rect,
-                                                    spr.blendmode)
-                    if spr.dirty == 1:
-                        spr.dirty = 0
-            _ret = list(_update)
-        else: # flip, full screen mode
+            self._draw_dirty_internal(self.spritedict, _rect,
+                                      self._spritelist,
+                                      _surf_blit, self.lostsprites)
+            _ret = list(self.lostsprites)
+        else:  # flip, full screen mode
             if _bgd is not None:
                 _surf_blit(_bgd, (0, 0))
-            for spr in _sprites:
-                if spr._visible:
-                    _old_rect[spr] = _surf_blit(spr.image,
-                                                spr.rect,
-                                                spr.source_rect,
-                                                spr.blendmode)
-            _ret = [_rect(_clip)] # return only the part of the screen changed
-
+            for spr in self._spritelist:
+                if spr.visible:
+                    self.spritedict[spr] = _surf_blit(spr.image,
+                                                      spr.rect,
+                                                      spr.source_rect,
+                                                      spr.blendmode)
+            _ret = [_rect(_clip)]  # return only the part of the screen changed
 
         # timing for switching modes
         # How may a good threshold be found? It depends on the hardware.
-        end_time = get_ticks()
+        end_time = pygame.time.get_ticks()
         if end_time-start_time > self._time_threshold:
             self._use_update = False
         else:
             self._use_update = True
 
-##        # debug
-##        print "               check: using dirty rects:", self._use_update
-
         # emtpy dirty rects list
-        _update[:] = []
+        self.lostsprites[:] = []
 
         # -------
         # restore original clip
-        _surf.set_clip(_orig_clip)
+        surface.set_clip(_orig_clip)
         return _ret
 
+    @staticmethod
```

Comment:
Nice improvement moving these to separate functions.

I haven't read through these two to see if they were refactored correctly however.